jcl/src/java.base/share/classes/java/lang/JFRHelpers.java

```
@@ -38,6 +40,162 @@ final class JFRHelpers {
+					case "settings":
+						settings = value;
+						break;
+					case "delay":
+						delay = convertToNanoSeconds(value);
+						break;
+					case "duration":
+						duration = convertToNanoSeconds(value);
+						break;
+					case "maxage":
+						maxAge = convertToNanoSeconds(value);
+						break;
+					case "maxsize":
+						maxSize = convertToBytes(value);
+						break;
+					case "dumponexit":
+						dumpOnExit = Boolean.valueOf(value);
+						break;
+					case "disk":
+						disk = Boolean.valueOf(value);
+						break;
+					}
+				}
+			}
+			if (null == settings) {
+				settings = "default";
+			}
+			/* Convert the string to a String array */
+			String[] settingsArray = new String[] { settings };
+
+			Method executeMethod = dcmdStartClass.getDeclaredMethod(
+				"execute",
+				String.class, //name
+				String[].class, //settings
+				Long.class, //delay
+				Long.class, //duration
+				Boolean.class, //disk
+				String.class, //path
+				Long.class, //maxAge
+				Long.class, //maxSize
+				Boolean.class, //dumpOnExit
+				Boolean.class //pathToGcRoots
+			);
+
+			executeMethod.setAccessible(true);
+			String results = (String) executeMethod.invoke(
+				dcmdStartInstance,
+				null,
+				settingsArray,
+				delay,
+				duration,
+				disk,
+				fileName,
+				maxAge,
+				maxSize,
+				dumpOnExit,
+				null
+			);
+			if (null != results) {
+				logJFR(results, 0, 2);
+			}
+			/*[ENDIF] JAVA_SPEC_VERSION == 11 */
+
+			/*[IF JAVA_SPEC_VERSION >= 17]*/
+			Method executeMethod = dcmdStartClass.getSuperclass().getDeclaredMethod("execute", String.class, String.class, char.class);
+			executeMethod.setAccessible(true);
+
+			String[] results = (String []) executeMethod.invoke(dcmdStartInstance, "internal", jfrCMDLineOption, ',');
+			if (null != results) {
+				for (String result : results) {
+					logJFR(result, 0, 2);
+				}
+			}
+			/*[ENDIF] JAVA_SPEC_VERSION >= 17 */
+		} catch (InvocationTargetException e) {
+			Throwable cause = e.getCause();
```

please do seomthing like:
```
           Throwable cause = e.getCause();
			e.printStackTrace();
			throw new InternalError(cause);
```

If the failure occurs early in initialization logging will not be initialized and the error will be hidden